docs/docsite/rst/dev_guide/developing_collections.rst

```diff
@@ -504,6 +510,95 @@ The build process for docs.ansible.com will know where to find the module docs.
 
+.. _testing_collections:
+
+Testing Collections
+===================
+
+The main tool for testing collections is ``ansible-test``, Ansible's testing tool described in :ref:`developing_testing`. This tool provides several compile and sanity checks, and allows to run unit and integration tests for plugins.
+
+``ansible-test`` must always be executed in the root directory of a collection. In the following, we will describe how to run ``ansible-test`` with Docker containers. This is the easiest way to use ``ansible-test`` as no special requirements have to be installed, and this is also the way it is run in Shippable both in Ansible's main repository and in the large community collections such as `community.general <https://github.com/ansible-collections/community.general/>`_ and `community.network <https://github.com/ansible-collections/community.network/>`_.
+
+Compile and Sanity Tests
+------------------------
+
+To run all compile and sanity tests, execute::
+
+    ansible-test sanity --docker default -v
+
+See :ref:`testing_compile` and :ref:`testing_sanity` for more information on the tests run. See the :ref:`full list of sanity tests <all_sanity_tests>` for details on he sanity tests and how to fix identified issues.
+
+Unit Tests
+----------
+
+Unit tests must be located in the ``tests/unit/plugins/`` directory. The tests for ``plugins/module_utils/foo/bar.py`` must be in ``tests/unit/plugins/module_utils/foo/test_bar.py`` or ``tests/unit/plugins/module_utils/foo/bar/test_bar.py``. For examples, see the `unit tests in community.general <https://github.com/ansible-collections/community.general/tree/master/tests/unit/>`_.
+
+All unit tests can be run as follows::
+
+    ansible-test units --docker default -v
+
+This will run all unit tests for all supported Python versions. To run them only for a specific Python version::
+
+    ansible-test units --docker default -v --python 3.6
+
+To run only a specific unit test::
+
+    ansible-test units --docker default -v --python 3.6 tests/unit/plugins/module_utils/foo/test_bar.py
+
+Python requirements can be specified in the ``tests/unit/requirements.txt`` file. See :ref:`testing_units` for more information, especially on fixture files.
+
+Integration Tests
+-----------------
+
+Integration tests must be located in the ``tests/integration/targets``` directory. For every module ``foo``, a directory of name ``foo`` will contain the tests for that module. For lookup plugins, the directory must be called ``lookup_foo``; for connection plugins, ``connection_foo``; for inventory plugins, ``inventory_foo``.
+
+There are two types of integration tests:
+
+* Ansible roles: they will be run with ``ansible-playbook`` and expect to validate various aspects of the module. They can depend on other integration tests (usually named ``prepare_bar`` or ``setup_bar``) to set-up required resources, such as installing required libraries or setting up server services.
```

Comment:
I got confused when reading this. Initially I thought that this was describing how to setup role testing in a collection, but after re-reading a few times I realised it was describing using roles to test a module. The use of 'bar' here was part of the confusion - perhaps it needs something to set the stage, like "'prepare_bar' to prepare a service named  'bar' in order to test module 'foo'."

Reply:
I've changed the text in the bracket to `(usually named ``prepare_bar`` or ``setup_bar``, which prepare a service or install a requirement named ``bar`` in order to test module ``foo``)`. Does that look better to you?